language/compiler/ir-to-bytecode/syntax/src/lexer.rs

```diff
@@ -148,6 +150,18 @@ impl<'input> Lexer<'input> {
         self.token = (start_offset, Token(tok, result), end_offset);
         Ok(())
     }
+
+    pub fn replace_tok(
```

Comment:
Can we call this "replace_token" instead? It would be more consistent with the existing "find_token" function (which is internal, but still....)